backend/src/main/service/CachingService.ts

```diff
@@ -0,0 +1,85 @@
+import { RedisClient } from 'redis';
+import { CachingObject } from '../data/CachingObject';
+import { Promotion } from '../entity/Promotion';
+import { PromotionDTO } from '../validation/PromotionValidation';
+
+// handles the caching for lat/lon values for restaurants with redis
+export class CachingService {
+  private client: RedisClient;
+
+  // sets redis client for class
+  constructor(client: RedisClient) {
+    this.client = client;
+  }
+
+  // formats promotionDTO for caching
+  async getValuesAndCache(promotion: PromotionDTO): Promise<boolean> {
+    const placeId: string = promotion.placeId;
+    const latitude: number = promotion.lat;
+    const longitude: number = promotion.lon;
+    const result = await this.cacheLatLonValues(placeId, latitude, longitude);
+    return result;
+  }
+
+  // caches the lat/lon values for a promotion's restaurant
+  // key is placeId and value is of the form { lat: x, lon: y }
+  // sets the cached value to expire in 30 days (limit imposed by Google Places API)
+  async cacheLatLonValues(
+    placeId: string,
+    latitude: number,
+    longitude: number
+  ): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      const value: CachingObject = { lat: latitude, lon: longitude };
+      this.client.setex(
+        placeId,
+        2592000, // 30 days represented in seconds
+        JSON.stringify(value),
+        (err: Error | null) => {
+          if (err) {
+            return reject(err);
+          }
+          return resolve(true);
+        }
+      );
+    });
+  }
```

Comment:
How come we are returning a `boolean`? It seems like this value is only used for tests and we never return `resolve(false)` or `reject(false)`. We could return `Promise<void>` since if it's successful we know it didn't fail with a promise reject.